src/sys/pthread.rs

```diff
@@ -36,3 +38,22 @@ pub fn pthread_kill<T: Into<Option<Signal>>>(thread: Pthread, signal: T) -> Resu
     let res = unsafe { libc::pthread_kill(thread, sig) };
     Errno::result(res).map(drop)
 }
+
+/// Obtain the name of the thread
+///
+/// On linux the name cannot exceed 16 length including null terminator
+#[cfg(all(target_os = "linux", not(target_env = "musl")))]
+pub fn pthread_getname_np(thread: Pthread) -> Result<String> {
+    let mut name = [0u8; 16];
+    unsafe { libc::pthread_getname_np(thread, name.as_mut_ptr() as _, name.len()) };
+    let cname = unsafe { CStr::from_ptr(name.as_ptr() as _) };
+    Ok(cname.to_owned().to_string_lossy().to_string())
+}
+
+/// Set the name of the thread
+#[cfg(all(target_os = "linux", not(target_env = "musl")))]
+pub fn pthread_setname_np(thread: Pthread, name: String) {
```

Comment:
You could be a little more generic here and accept any `T: Into<Vec<u8>>`.  That way the caller could provide, for example, a byte string `b"nix-rust"`.